src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -232,45 +188,21 @@ class PythonSetup(Subsystem):
             this option, you can override each resolve to use certain interpreter
             constraints, such as `{'data-science': ['==3.8.*']}`.
 
+            Warning: this does NOT impact the interpreter constraints used by targets within the
+            resolve, which is instead set by the option `[python.interpreter_constraints` and the
+            `interpreter_constraints` field. It only impacts how the lockfile is generated.
+
             Pants will validate that the interpreter constraints of your code using a
             resolve are compatible with that resolve's own constraints. For example, if your
             code is set to use ['==3.9.*'] via the `interpreter_constraints` field, but it's
-            also using a resolve whose interpreter constraints are set to ['==3.7.*'], then
+            using a resolve whose interpreter constraints are set to ['==3.7.*'], then
             Pants will error explaining the incompatibility.
 
             The keys must be defined as resolves in `[python].resolves`.
             """
         ),
         advanced=True,
     )
-    no_binary = StrListOption(
```

Comment:
Moved to lower in the file. We have a block of about 10 lockfile related options, and this was in the middle.